messagecatalog/spec.md

```diff
@@ -0,0 +1,1475 @@
+# CNCF Message and Event Catalog API
+
+## Abstract
+
+This specification defines a data model, document format, and API for storing,
+organizing, and accessing catalogs of message definitions for use with event and
+messaging infrastructures using common, standardized event and message
+information models like those of CNCF CloudEvents, CNCF NATS, OASIS AMQP, OASIS
+MQTT and others.
+
+## Status of this document
+
+This document is a working draft.
+
+## Terminology
+
+A message is a data structure for transferring and dispatching
+metadata-annotated information through some asynchronous communication framework
+or infrastructure. An event is a kind of message.
+
+A producer produces messages and publishes them. A subscriber subscribes to
+published messages based on some criteria. A consumer is the delivery
+destination for such subscriptions and handles the delivered information.
+
+## 1. Introduction
+
+Message and event catalogs are useful tools at design and development time
+because they allow interactive discovery of the kinds of messages that are
+available for subscription from a system and to find out, for instance, which
+event sources raise them. A chosen message definition catalog entry can serve as
```

Comment:
Couple of questions:
1 - are people going to be looking for "messages" or "events" ? I would guess it's probably "events" and then they would use the related "message" to understand the extra metadata around it.
2 - does this spec assume "structured cloudEvents"? I'm asking because (unless I missed it) it seems like the message definitions assume all properties are in the Body. And just to complete things... in the "binary CE" cases, the schema of the Body would be the dataschema, right?